haystack/preview/components/websearch/serper_dev.py

```diff
@@ -0,0 +1,161 @@
+import json
+import logging
+from typing import Dict, List, Optional, Any
+
+import requests
+
+from haystack.preview import Document, component, default_from_dict, default_to_dict
+from haystack.preview.components.websearch.utils import score_results
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class SerperDev:
+    """
+    Search engine using SerperDev API. See the [Serper Dev website](https://serper.dev/) for more details.
+    """
+
+    def __init__(
+        self,
+        api_key: str,
+        top_k: Optional[int] = 10,
+        allowed_domains: Optional[List[str]] = None,
+        search_params: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        :param api_key: API key for the SerperDev API.
+        :param top_k: Number of documents to return.
+        :param allowed_domains: List of domains to limit the search to.
+        :param search_params: Additional parameters passed to the SerperDev API.
+        For example, you can set 'num' to 20 to increase the number of search results.
+        """
+        if api_key is None:
+            raise ValueError("API key for SerperDev API must be set.")
+        self.api_key = api_key
+        self.top_k = top_k
+        self.allowed_domains = allowed_domains
+        self.search_params = search_params if search_params else {}
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
+        """
+        return default_to_dict(
+            self,
+            api_key=self.api_key,
+            top_k=self.top_k,
+            allowed_domains=self.allowed_domains,
+            search_params=self.search_params,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "SerperDev":
+        """
+        Deserialize this component from a dictionary.
+        """
+        return default_from_dict(cls, data)
+
+    @component.output_types(documents=List[Document])
+    def run(
+        self,
+        query: str,
+        allowed_domains: Optional[List[str]] = None,
+        top_k: Optional[int] = None,
+        search_params: Optional[Dict[str, Any]] = None,
```

Comment:
I'd keep the minimal amount of inputs. Are we sure we need these? I see they're already set in init. 😕

Reply:
Ok, we can later add back in  `search_params: Optional[Dict[str, Any]] = None` if needed